### DISCUSSION TITLE
Proposal to add new path to REST API

### DISCUSSION THREAD

**Discussion by jeffreyscarpenter** [stargate/stargate#2288] [RFCs]
We propose to add a new path to the REST v2 API as a simplified way to execute CQL queries:

- The operation will be a POST to the `/v2/cql` endpoint. 
- This operation will accept a CQL query string as the message body (with no adornment)
- Results will be returned using the same JSON format as the GET `/v2/keyspaces/<ks>/tables` operations
- The new operation will be added to Swagger docs and the stargate.io/docs site
- We could potentially make the exposure of this additional operation optional via a property, and either default to enabled or disabled based on user feedback to this discussion.

This is targeted as an improvement for enabling web-based applications, but could have other uses. An initial implementation is proposed under PR #2266.

**Comment by jeffreyscarpenter** [comment]
This would be added to both the Stargate v1 (`v1` branch) and v2 (default `main` branch) implementations since we are actively maintaining both at this time.

**Reply by ivansenic** [reply]
I would argue that adding this to `v1` should not be a priority. Would only target the `main` branch.

**Reply by jeffreyscarpenter** [reply]
Ok, this is fine.

**Comment by polandll** [comment]
Will you want it documented beyond the swagger ui?

**Reply by jeffreyscarpenter** [reply]
> Will you want it documented beyond the swagger ui?

Yes, we should. Especially because we're putting a feature flag of some type around it and will need to describe how to enable it.

**Comment by jeffreyscarpenter** [comment]
This was added in the Stargate v2.0.4 release.
